File: backend/src/main/java/com/hireorbit/hire/orbit/entity/JobApplication.java
package com.hireorbit.hire.orbit.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "job_applications")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class JobApplication {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Title is required")
    private String title;

    @NotBlank(message = "Company is required")
    private String company;

    @NotBlank(message = "Status is required")
    private String status;

    private LocalDate appliedDate;
    private LocalDate deadline;
    private String notes;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    @JsonIgnore // prevents exposing password & lazy-load issues
    private User user;
}

File: frontend/lib/features/jobs/models/job_application.dart
class JobApplication {
  final int? id; 
  final String title;
  final String company;
  final String status;
  final DateTime? appliedDate;
  final DateTime? deadline;
  final String? notes;

  JobApplication({
    this.id,
    required this.title,
    required this.company,
    required this.status,
    this.appliedDate,
    this.deadline,
    this.notes,
  });

  factory JobApplication.fromJson(Map<String, dynamic> json) {
    return JobApplication(
      id: json['id'],
      title: json['title'],
      company: json['company'],
      status: json['status'],
      appliedDate: json['appliedDate'] != null
          ? DateTime.parse(json['appliedDate'])
          : null,
      deadline: json['deadline'] != null ? DateTime.parse(json['deadline']) : null,
      notes: json['notes'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'company': company,
      'status': status,
      'appliedDate': appliedDate?.toIso8601String(),
      'deadline': deadline?.toIso8601String(),
      'notes': notes,
    };
  }
}